### PR TITLE
Fix 2264 learning stone & Learning skill

### PR DESCRIPTION
**Pull request by vmarkovtsev**
Try to fix http://bugs.vcmi.eu/view.php?id=2264

### CODE REVIEW

File: lib/mapObjects/CRewardableObject.h [hunk 1]
```diff
@@ -104,9 +104,10 @@ class DLL_LINKAGE CRewardInfo
 	/// if set to true, object will be removed after granting reward
 	bool removeObject;
 
-	/// Generates list of components that describes reward
-	virtual void loadComponents(std::vector<Component> & comps) const;
-	Component getDisplayedComponent() const;
+	/// Generates list of components that describes reward for a specific hero
+	virtual void loadComponents(std::vector<Component> & comps,
+	                            const CGHeroInstance * h) const;
```
Comment (DjWarmonger) [inline lib/mapObjects/CRewardableObject.h]: You really don't need o break lines like that in the age of HD monitors ;)

Comment (vmarkovtsev) [inline lib/mapObjects/CRewardableObject.h]: Have you ever done a 3-way merge with 120-char lines?

Comment (vmarkovtsev) [inline lib/mapObjects/CRewardableObject.h]: To be more precise, here is the motivation: https://google.github.io/styleguide/cppguide.html#Line_Length

Comment (DjWarmonger) [inline lib/mapObjects/CRewardableObject.h]: For me it's quite the opposite with one large code window in Visual Studio. Once I had to work at split-screen and it was a nightmare.

**Pull request merged**